crypto/rand/rand_unix.c

```diff
@@ -313,25 +370,38 @@ size_t rand_pool_acquire_entropy(RAND_POOL *pool)
 
 #   ifdef OPENSSL_RAND_SEED_DEVRANDOM
     bytes_needed = rand_pool_bytes_needed(pool, 1 /*entropy_factor*/);
-    if (bytes_needed > 0) {
-        static const char *paths[] = { DEVRANDOM, NULL };
-        FILE *fp;
-        int i;
+    {
+        size_t i;
 
-        for (i = 0; paths[i] != NULL; i++) {
-            if ((fp = fopen(paths[i], "rb")) == NULL)
-                continue;
-            setbuf(fp, NULL);
+        for (i = 0; bytes_needed > 0 && i < OSSL_NELEM(devrandom_paths); i++) {
+            FILE *fp = devrandom_files[i];
+
+            if (fp == NULL) {
```

Comment:
I would like to see a call to fstat(), and check S_ISCHR(), to see if the file is still open and still the type of file we expect.

Reply:
Again, I'm happy to change this.  Should we trust the configuration setting or should we check?

Reply:
The reason I'm asking this is that I know that there is software that part as becoming a daemon will close all files. So they might have closed our file without us knowing about it, and it's very likely that something will reuse that file descriptor at a later point, say a network socket or log file. I want to avoid reading from an unexpected file. I guess we could cache the whole struct stat and check that some of the fields match.

Reply:
That's a good enough reason for me.  I'll add the check.